### DISCUSSION TITLE
asUploadButton()'s onClick prop is missing, intentional?

### DISCUSSION THREAD

**Discussion by AJLemos** [rpldy/react-uploady#368] [Q&A]
We are currently using the `asUploadButton()` HOC for a custom upload button.

The docs for this HOC state:

> The props that the new component (returned by HOC) are the same as described in the [Upload Button Props table](https://react-uploady.netlify.app/docs/api/components/uploadButton/#props). In order to pass props to the wrapped component, use the `extraProps` prop.

However, `asUploadButton` returns `UploadButtonProps`, which does not include an `onClick` prop ([seen here](https://github.com/rpldy/react-uploady/blob/master/packages/ui/upload-button/types/index.d.ts)). The docs indicate the props should match the `<UploadButton>` component, which does include this prop ([seen here](https://github.com/rpldy/react-uploady/blob/master/packages/ui/upload-button/src/types.js)).

This creates a type checking error when using these props (seen below).

Is the omission of the `onClick` prop intentional? Should the docs indicate this? Or should `onClick` be included in the `ButtonProps` interface?

We need to call a side-effect on this click to track the event in our analytics. Here is simplified example of our custom button:

```jsx
// @ts-check
import { asUploadButton } from '@rpldy/upload-button';

const CallToAction = asUploadButton((/** @type {UploadButtonProps} */ props) => {
  return (
    <button
      {...props}
      onClick={() => {
        trackFileBrowseClick();

        // @ts-expect-error: The `UploadButtonProps` type  does not include an
        // `onClick` prop, but it exists and has to be called on click to show
        // the file picker.
        props.onClick();
      }}
    >
      Click Here To Upload
    </button>
  };
});

function trackFileBrowseClick() {
  // side-effect code
}

/** @typedef {import('@rpldy/upload-button').UploadButtonProps} UploadButtonProps */
```

I also verified that the same type error occurs when adding an `onClick` prop to the `<UploadButton>` component.

```jsx
// @ts-check
import { asUploadButton } from '@rpldy/upload-button';

function CustomUploadButton() {
  return (
    // @ts-expect-error
    <UploadButton onClick={() => {}}>
      Click Here To Upload
    </UploadButton>
  );
}
```

I see in [this discussion](https://github.com/rpldy/react-uploady/discussions/198) that using `showFileUpload()` might actually simplify our use case. But I was curious if the docs or the interface need to be updated.

**Comment by yoavniran** [comment]
Hey @AJLemos 
This indeed seems like an oversight on my part. I will add (#369) `onClick` to the `ButtonProps` interface. 
thanks!

**Reply by AJLemos** [reply]
Got it! I wasn't sure if this was intentional or not, which is why I didn't open an issue right away. Thanks!

**Comment by yoavniran** [comment]
@AJLemos version 0.18.3 released and available now with the updated definitions
